### PR TITLE
os: add configuring dns document

### PR DESCRIPTION
**Pull request by mischief**
/cc @crawford @robszumski

### CODE REVIEW

File: os/configuring-dns.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+See [Network configuration with networkd]({{site.baseurl}}/docs/os/network-config-with-networkd) for more information on `systemd-networkd`.
+
+## Using a local DNS cache
+
+Systemd-resolved includes a DNS stub resolver.
+To use it as a DNS cache, you must enable it via [nsswitch.conf][nsswitch.conf] by adding `resolv` to the `hosts` section.
+
+Here is an example cloud-config snippet to do that:
+
+```yaml
```
Comment (robszumski) [inline os/configuring-dns.md]: Validated this FYI

File: os/configuring-dns.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# DNS configuration
+
+By default, DNS resolution on CoreOS is handled through `/etc/resolv.conf`, which is a symlink to `/run/systemd/resolve/resolv.conf`.
+This file is managed by [systemd-resolved][systemd-resolved].
+Normally, `systemd-resolved` gets DNS IP addresses from [systemd-networkd][systemd-networkd], either via DHCP or static configuration.
+DNS IP addresses can also be set via `systemd-resolved`'s [resolved.conf][resolved.conf].
+See [Network configuration with networkd]({{site.baseurl}}/docs/os/network-config-with-networkd) for more information on `systemd-networkd`.
```
Comment (robszumski) [inline os/configuring-dns.md]: We're rewriting these links now when they get sucked in, so you can probably make this link work for people viewing this on Github. I think it should be:

`[Network configuration with networkd](network-config-with-networkd.md)`

File: os/configuring-dns.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# DNS configuration
```
Comment (robszumski) [inline os/configuring-dns.md]: title case: `DNS Configuration`

File: os/configuring-dns.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+By default, DNS resolution on CoreOS is handled through `/etc/resolv.conf`, which is a symlink to `/run/systemd/resolve/resolv.conf`.
+This file is managed by [systemd-resolved][systemd-resolved].
+Normally, `systemd-resolved` gets DNS IP addresses from [systemd-networkd][systemd-networkd], either via DHCP or static configuration.
+DNS IP addresses can also be set via `systemd-resolved`'s [resolved.conf][resolved.conf].
+See [Network configuration with networkd]({{site.baseurl}}/docs/os/network-config-with-networkd) for more information on `systemd-networkd`.
+
+## Using a local DNS cache
+
+Systemd-resolved includes a DNS stub resolver.
```
Comment (robszumski) [inline os/configuring-dns.md]: `systemd-resolved`, right?

Comment (robszumski) [inline os/configuring-dns.md]: What is a stub resolver? Can we explain that more if its important?

Comment (mischief) [inline os/configuring-dns.md]: i echo'd what the systemd-resolved manual says. maybe best to drop 'stub' and simply claim it is a caching dns resolver.

**Comment by mischief** [comment]
@robszumski ptal

**Comment by robszumski** [comment]
LGTM

**Pull request merged**